pyteal/ast/scratch.py

```diff
@@ -9,12 +11,30 @@
 class ScratchSlot:
     """Represents the allocation of a scratch space slot."""
 
-    slotId = 0
+    # Unique identifier for the compiler to automatically assign slots
+    # The id field is used by the compiler to map to an actual slot in the source code
+    # Slot ids under 256 are manually reserved slots
+    nextSlotId = 256 
 
-    def __init__(self):
-        self.id = ScratchSlot.slotId
-        ScratchSlot.slotId += 1
+    def __init__(self, requestedSlotId: int = None):
+        """Initializes a scratch slot with a particular id
 
+        Args:
+            requestedSlotId (optional): A scratch slot id that the compiler must store the value. 
+            This id may be a Python int in the range [0-256).       
+        """
+        if requestedSlotId is None:
+            self.id = ScratchSlot.nextSlotId
+            ScratchSlot.nextSlotId += 1
+            self.isReservedSlot = False
+        else:
+            # TODO: Is there a way to check whether the user hasn't alloted more than 
```

Comment:
To answer the TODO, no I don't think that's possible here. Since multiple programs can be defined in the same Python runtime, you can't figure out if the slots are exhausted until compile time.

Reply:
Is this TODO meant to get committed?